gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/FieldValue.java

```diff
@@ -0,0 +1,207 @@
+package com.google.gcloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.api.client.util.Data;
+import com.google.api.client.util.Lists;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Google BigQuery Table Field Value class. Objects of this class represent values of a BigQuery
+ * Table Field. A list of values forms a {@link TableRow}. Tables rows can be get as the result of
+ * a query or when listing table data.
+ */
+public class FieldValue implements Serializable {
+
+  private static final int MICROSECONDS = 1000000;
+  private static final long serialVersionUID = 469098630191710061L;
+
+  private final Kind kind;
+  private final Object value;
+
+  /**
+   * The field value's kind, giving information on the field's content type.
+   */
+  public enum Kind {
+    /**
+     * A primitive field value. A {@code FieldValue} has type {@link #PRIMITIVE} when the
+     * corresponding field has type {@link Field.Type#bool()}, {@link Field.Type#string()}
+     * {@link Field.Type#floatingPoint()}, {@link Field.Type#integer()},
+     * {@link Field.Type#timestamp()} or the value is set to {@code null}.
+     */
+    PRIMITIVE,
+
+    /**
+     * A {@code FieldValue} for a field with {@link Field.Mode#REPEATED} mode.
+     */
+    REPEATED,
+
+    /**
+     * A {@code FieldValue} for a field of type {@link Field.Type#record(Field...)}.
+     */
+    RECORD
+  }
+
+  FieldValue(Kind kind, Object value) {
+    this.kind = kind;
+    this.value = value;
+  }
+
+  /**
+   * Returns the kind of this Field Value.
+   *
+   * @return {@link Kind#PRIMITIVE} if the value is of primitive type ({@link Field.Type#bool()},
+   *     {@link Field.Type#string()}, {@link Field.Type#floatingPoint()},
+   *     {@link Field.Type#integer()}, {@link Field.Type#timestamp()}) or is {@code null}. Returns
+   *     {@link Kind#REPEATED} if the corresponding field has ({@link Field.Mode#REPEATED}) mode.
+   *     Returns {@link Kind#RECORD} if the corresponding field has
+   *     {@link Field.Type#record(Field...)} type.
+   */
+  public Kind kind() {
+    return kind;
+  }
+
+  /**
+   * Return this field's value as an {@link Object}.
+   */
+  public Object value() {
+    return value;
+  }
+
+  /**
+   * Return this field's value as an {@link String}.
+   */
```

Comment:
Should this javadoc have: "This method should only be used if the corresponding field has {@link Field.Type#string}" warning and `@throws` statement?